node/node.go

```diff
@@ -143,12 +149,17 @@ func errorHandler(_err *C.struct_SalticidaeCError, fatal C.bool, asyncID C.int32
 func (n *Node) initNetlib() error {
 	// Create main event context
 	n.EC = salticidae.NewEventContext()
+	n.TCall = salticidae.NewThreadCall(n.EC)
 
-	// Set up interrupt signal and terminate signal handlers
-	evInt := salticidae.NewSigEvent(n.EC, salticidae.SigEventCallback(C.onTerm), nil)
-	evInt.Add(salticidae.SIGINT)
-	evTerm := salticidae.NewSigEvent(n.EC, salticidae.SigEventCallback(C.onTerm), nil)
-	evTerm.Add(salticidae.SIGTERM)
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, os.Interrupt)
```

Comment:
How about creating signal callback func ? this can be used across application. just a suggestion.
```
// invoke utils.go
NewNotifierOnSignal(func(s os.Signal) {
            n.TCall.AsyncCall(salticidae.ThreadCallCallback(C.onTerm), nil)
})
```

```
// utils.go
type  SignalCallback func(s os.Signal)

func NewNotifierOnSignal(callback SignalCallback, sig ...os.Signal) {
    if callback != nil && len(sig) > 0 {
        go func() {
            sigChan := make(chan os.Signal, 1)
            defer signal.Stop(sigChan)

            signal.Notify(sigChan, sig...)
            select {
            case s := <-sigChan:
                callback(s)
            }
        }()
    }
}
```

Reply:
Updated signal handling into utils.